README.md

```diff
@@ -348,10 +348,46 @@ The generated parameter value for the nested map example can then be accessed wi
 
 ### Use generated struct in Cpp
 The generated header file is named based on the target library name you passed as the first argument to the cmake function.
-If you specified it to be `turtlesim_parameters` you can then include the generated code with `#include "turtlesim_parameters.hpp"`.
+If you specified it to be `turtlesim_parameters` you can then include the generated code with `#include <turtlesim/turtlesim_parameters.hpp>`.
 ```c++
-#include "turtlesim_parameters.hpp"
+#include <turtlesim/turtlesim_parameters.hpp>
 ```
+
+Note that this header can also be used from another package:
+```cmake
+cmake_minimum_required(VERSION 3.8)
+project(my_other_package)
+
+
+include(GNUInstallDirs)
+
+# find dependencies
+find_package(ament_cmake REQUIRED)
+find_package(turtelsim REQUIRED)
+
+add_library(my_lib src/my_lib.cpp)
+target_include_directories(my_lib PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+    $<INSTALL_INTERFACE:include>)
+target_link_libraries(my_lib PUBLIC turtlesim::turtlesim_parameters)
+
+#############
+## Install ##
+#############
+
+install(DIRECTORY include/${PROJECT_NAME}/ DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/${PROJECT_NAME})
+
+install(TARGETS my_lib
+  EXPORT ${PROJECT_NAME}Targets
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  RUNTIME DESTINATION lib/${PROJECT_NAME})
+
+ament_export_targets(${PROJECT_NAME}Targets HAS_LIBRARY_TARGET)
```

Comment:
The equivalent of these two instructions have to be added in the library, which exports the turtlesim_parameters. Please add this to this section here (it is not part of the [minimum CMakeLists snippet](https://github.com/PickNikRobotics/generate_parameter_library/blob/main/README.md#add-parameter-library-generation-to-project))

Reply:
would there be any drawbacks?

Reply:
The only effect would be to export the generated header target even if you don't use it outside of your package, which sounds fine to me (same thing with exported dependencies). The main advantage is to group everything under the generate_parameter_library cmake command, which makes it very easy to use.

Reply:
again, I vote for the simplest version to use ;) but I'd understand if someone wants to keep the created artifacts as small as possible

Reply:
Done. To achieve this, I transformed the function into a macro, so that I don't need to globalize every cmake variable set in the function.